src/wrapmime.rs

```diff
@@ -108,48 +298,77 @@ pub fn get_mime_transfer_encoding(mime: *mut Mailmime) -> Option<libc::c_int> {
 pub fn decode_dt_data(
     mime_data: *mut mailmime_data,
     mime_transfer_encoding: libc::c_int,
-) -> Result<(*mut libc::c_char, libc::size_t), Error> {
+) -> Result<Vec<u8>, Error> {
```

Comment:
this and the below code now uses the code that was in dc_mimeparser -- the below deleted code might have produced a leak with decoded_data -- or worse?  in any case it's better to return a Vec, even if it means a copy.